react-game/src/components/AllTrials.js

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect } from 'react';
+import StartButton from './Buttons/StartButton';
+import classes from './AllTrials.module.css';
+import TrialRow from './TrialRow/TrialRow';
+
+function AllTrials() {
+
+    //STATES & HOOKS-------------------------------------------------------------------
+    const [secretCode, setSecretCode] = useState("0000");
+    const [selectedIndex, setSelectedIndex] = useState(0); //this state holds the current enabled row  
+    const [gameStatus, setGameStatus] = useState(0); // 0 - means the game has not started yet 
+    // 1 - means the game is active
+    // 2 - means the game finished successfully
+    // 3 - means the game finished with a failure 
+    const [resultText, setResultText] = useState(""); //the result text that indicates if the user won or lost the game
+    const [clearScreen, setClearScreen] = useState(true); //to reset the screen when the user starts the game again
+
+
+    useEffect(() => { //control the status of the game
+        if (gameStatus === 1) { //this restarts the game and clears the screen
+            setResultText("");
+            setSecretCode(generateRandomNumber(1000, 9999)); //set a new random secret code
+            setSelectedIndex(1); //to enable the first row again
+            setClearScreen(true);
+        }
+        else if (gameStatus === 2) {
+            setResultText("Congrats, You won the game!");
+        }
+        else if (gameStatus === 3) {
+            setResultText("Game Over, You lost the game :(");
+        }
+    }, [gameStatus]);
+
+
+    //FUNCTIONS-----------------------------------------------------------------
+    function generateRandomNumber(min, max) { //generate number between min (inclusive) and max (inclusive)
```

Comment:
please use arrow function instead of normal function in react components

Reply:
done